eslint-prettier-config.sh

```diff
@@ -5,37 +5,166 @@ GREEN='\033[1;32m'
+  if [[ $REPLY =~ ^[Nn]$ ]]; then
+    echo ">>>>> Skipping ESLint config"
+    skip_eslint_setup="true"
+  fi
+fi
+
+finished=false
+while ! $finished; do
+  echo
+  read -p "????? What max line length do you want to set for ESLint and Prettier (Recommendations: 80 or 100 or 120)? "
+  if [[ $REPLY =~ ^[0-9]{2,3}$ ]]; then
+    max_len_val=$REPLY
+    finished=true
+    echo
+  else
+    echo "!!!!! Please choose a max length of two or three digits, e.g. 80 or 100 or 120"
+  fi
+done
+
+echo "????? What style of trailing commas do you want to enforce with prettier?"
+echo ">>>>> See https://prettier.io/docs/en/options.html#trailing-commas for more details."
+select trailing_comma_pref in "none" "es5" "all"; do
+  case $trailing_comma_pref in
+    none ) break;;
+    es5 ) break;;
+    all ) break;;
+  esac
+done
+echo
+
+if [ -f ".prettierrc.js" -o -f "prettier.config.js" -o -f ".prettierrc.yaml" -o -f ".prettierrc.yml" -o -f ".prettierrc.json" -o -f ".prettierrc.toml" -o -f ".prettierrc" ]; then
+  echo "!!!!! Existing Prettier config file(s) found"
+  ls -a | grep "prettier*" | xargs -n 1 basename
+  echo
+  echo ">>>>> CAUTION: The configuration file will be resolved starting from the location of the file being formatted, and searching up the file tree until a config file is (or isn't) found. https://prettier.io/docs/en/configuration.html"
+  echo
+  read -p  "????? Write .prettierrc${config_extension} (Y/n)? "
+  if [[ $REPLY =~ ^[Nn]$ ]]; then
+    echo ">>>>> Skipping Prettier config"
+    skip_prettier_setup="true"
+  fi
+  echo
+fi
+
+
+# Pefrom config
+# -------------
+echo
 echo -e "${GREEN}Starting Style Formatting Configuration... ${NC}"
 
+echo
 echo -e "1/5 ${LCYAN}Local ESLint & Prettier Installation... ${NC}"
-npm install -D eslint@5.6.0 prettier
+echo
+$pkg_cmd -D eslint prettier
+
 
+echo
+echo
 echo -e "2/5 ${YELLOW}Airbnb Configuration Installation... ${NC}"
-npm install -D eslint-config-airbnb eslint-plugin-jsx-a11y eslint-plugin-import eslint-plugin-react babel-eslint
+echo
+$pkg_cmd -D eslint-config-airbnb eslint-plugin-jsx-a11y eslint-plugin-import eslint-plugin-react babel-eslint
 
+
+echo
+echo
 echo -e "3/5 ${LCYAN}Disabling ESLint Formatting... ${NC}"
-npm install -D eslint-config-prettier eslint-plugin-prettier eslint-plugin-flowtype@2.50.3 eslint-config-react-app
+echo "See https://github.com/prettier/eslint-config-prettier for more details."
+echo
+$pkg_cmd -D eslint-config-prettier eslint-plugin-prettier eslint-plugin-flowtype
+
 
-echo -e "4/5 ${YELLOW}Creating ESLint JSON... ${NC}"
-touch .eslintrc.json
+if [ "$skip_eslint_setup" == "true" ]; then
+  break
+else
+  echo
+  echo
+  echo -e "4/5 ${YELLOW}Creating custom .eslintrc${config_extension} config...${GREEN}Done! ${NC}"
+  > ".eslintrc${config_extension}" # truncates existing file (or creates empty)
 
-echo '{
-  "extends": ["airbnb", "prettier", "react-app", "plugin:prettier/recommended"],
-  "plugins": ["prettier"],
+  echo ${config_opening}'
```

Comment:
I got an error in VS Code when I chose the eslintrc to be in .json format. The error stated that it couldn't read the file because comments are not allowed in .json files.

fyi: I tested the script when I chose the eslintrc file to be in .js format and it worked perfectly.

Reply:
@paulolramos I noticed this, too. And while it does seem that ESLint complains about the comments, ironically their docs also say they support them in the config 🤔 I'm not sure what to do here, but learning towards ignoring that file until ESLint figures out that it's ok to do what they said they allow...

https://eslint.org/docs/user-guide/configuring#configuration-file-formats

![image](https://user-images.githubusercontent.com/4521/50918334-8f1dd600-1405-11e9-9305-1e0d482136ac.png)

Reply:
@jondkinney That sounds good. We can just ignore that file until comments are supported. Would that mean the `.eslintrc.json` would be taken off as an option or give the user a warning that by choosing that option it might not work?